src/memcached/stats_collecting_client.go

```diff
@@ -0,0 +1,80 @@
+package memcached
+
+import (
+	"github.com/bradfitz/gomemcache/memcache"
+	stats "github.com/lyft/gostats"
+)
+
+type statsCollectingClient struct {
+	c Client
+
+	multiGetSuccess  stats.Counter
+	multiGetError    stats.Counter
+	incrementSuccess stats.Counter
+	incrementMiss    stats.Counter
+	incrementError   stats.Counter
+	addSuccess       stats.Counter
+	addError         stats.Counter
+	addNotStored     stats.Counter
+	keysRequested    stats.Counter
+	keysFound        stats.Counter
+}
+
+func CollectStats(c Client, scope stats.Scope) Client {
+	return statsCollectingClient{
+		c:                c,
+		multiGetSuccess:  scope.NewCounterWithTags("multiget", map[string]string{"code": "success"}),
+		multiGetError:    scope.NewCounterWithTags("multiget", map[string]string{"code": "error"}),
+		incrementSuccess: scope.NewCounterWithTags("increment", map[string]string{"code": "success"}),
+		incrementMiss:    scope.NewCounterWithTags("increment", map[string]string{"code": "miss"}),
+		incrementError:   scope.NewCounterWithTags("increment", map[string]string{"code": "error"}),
+		addSuccess:       scope.NewCounterWithTags("add", map[string]string{"code": "success"}),
+		addError:         scope.NewCounterWithTags("add", map[string]string{"code": "error"}),
+		addNotStored:     scope.NewCounterWithTags("add", map[string]string{"code": "not_stored"}),
+		keysRequested:    scope.NewCounter("keys_requested"),
+		keysFound:        scope.NewCounter("keys_found"),
```

Comment:
I don't see any tests for these? Can you make sure they are all covered? Thank you.

/wait

Reply:
keys_requested and keys_found are tested during the tests for GetMulti. I've added one more test just now so there are tests where keys_found and keys_requested are either 1 or 2.

Reply:
Sorry, missed this. Awesome, thanks!